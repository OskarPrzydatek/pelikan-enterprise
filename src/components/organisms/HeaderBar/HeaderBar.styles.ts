import styled from 'styled-components';

export const HeaderBar = styled.header`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 ${({ theme }) => theme.spaces.xl};
  box-shadow: 0px 4px 4px 0px ${({ theme }) => theme.colors.shadowBlack};
`;
