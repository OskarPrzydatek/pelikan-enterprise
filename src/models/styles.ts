import {
  DefaultTheme,
  FlattenInterpolation,
  ThemeProps,
} from 'styled-components';

export interface ICSS {
  css?: FlattenInterpolation<ThemeProps<DefaultTheme>>;
}
