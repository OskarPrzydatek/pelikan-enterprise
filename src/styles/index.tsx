export * from './layouts';
export { theme } from './theme';
