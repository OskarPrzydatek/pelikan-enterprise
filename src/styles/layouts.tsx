import styled, { createGlobalStyle } from 'styled-components';

export const GlobalStyleLayout = createGlobalStyle`
  * {
		box-sizing: border-box;
    outline: none;
	} 

  body {
    margin: 0;
    padding: 0;
  }

  #root {
    height: 100vh;
    display: grid;
    grid-template-rows: .5fr 8fr .5fr;
  }

	a {
		color: inherit;
		text-decoration: none;
	}

	ul {
		padding: 0;
    margin: 0;
		list-style-type: none;
	}
`;

export const MainLayout = styled.main`
  padding: ${({ theme }) => theme.spaces.l};
`;

export const PageLayout = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;
