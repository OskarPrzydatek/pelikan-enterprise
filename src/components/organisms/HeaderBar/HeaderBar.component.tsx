import { ProfilePicture } from '~/components/molecules';

import { HamburgerMenu } from '../HamburgerMenu/HamburgerMenu.component';

import * as S from './HeaderBar.styles';

export const HeaderBar: React.FC = () => {
  return (
    <S.HeaderBar>
      <HamburgerMenu />
      <ProfilePicture label="Witaj User" />
    </S.HeaderBar>
  );
};
