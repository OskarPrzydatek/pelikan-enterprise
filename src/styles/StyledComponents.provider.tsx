import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from './layouts';
import { theme } from './theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};
