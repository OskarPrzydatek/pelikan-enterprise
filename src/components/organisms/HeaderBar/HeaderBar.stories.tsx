import { Meta, StoryObj } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';

import { StyledComponentsProvider } from '~/providers';

import { HeaderBar } from './HeaderBar';

const meta: Meta<typeof HeaderBar> = {
  title: 'Organisms/HeaderBar',
  component: HeaderBar,
};

export default meta;
type Story = StoryObj<typeof HeaderBar>;

export const HeaderBarStory: Story = {
  render: () => (
    <BrowserRouter>
      <StyledComponentsProvider>
        <HeaderBar />
      </StyledComponentsProvider>
    </BrowserRouter>
  ),
};
