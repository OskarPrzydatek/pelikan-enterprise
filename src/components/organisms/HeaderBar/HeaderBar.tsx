import { ProfilePicture } from '~/components/molecules';

import { HamburgerMenu } from './HamburgerMenu/HamburgerMenu';
import * as S from './HeaderBar.styles';

export const HeaderBar: React.FC = () => {
  return (
    <S.HeaderBar>
      <HamburgerMenu />
      <ProfilePicture label="Witaj UÅ¼ytkowniku" />
    </S.HeaderBar>
  );
};
