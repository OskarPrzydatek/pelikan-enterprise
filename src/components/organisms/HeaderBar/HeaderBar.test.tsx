import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { describe, expect, test } from 'vitest';

import { StyledComponentsProvider } from '~/styles';

import { HeaderBar } from './HeaderBar.component';

describe('HeaderBar', () => {
  test('component snapshot', () => {
    const view = render(
      <BrowserRouter>
        <StyledComponentsProvider>
          <HeaderBar />
        </StyledComponentsProvider>
      </BrowserRouter>
    );
    expect(view).toMatchSnapshot();
  });
});
