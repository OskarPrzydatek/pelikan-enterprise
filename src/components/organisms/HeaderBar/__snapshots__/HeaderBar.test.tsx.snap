// Vitest Snapshot v1

exports[`HeaderBar > component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-bqWxrE gOWFEp"
      >
        <button
          class="sc-eDvSVe bcqqiU"
          data-testid="side-menu-hamburger-icon"
        >
          <svg
            data-testid="hamburger-icon"
            fill="none"
            height="15"
            viewBox="0 0 45 23"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              stroke="#013059"
              stroke-width="3"
              x2="45"
              y1="21.5"
              y2="21.5"
            />
            <line
              stroke="#013059"
              stroke-width="3"
              x2="45"
              y1="11.5"
              y2="11.5"
            />
            <line
              stroke="#013059"
              stroke-width="3"
              x2="45"
              y1="1.5"
              y2="1.5"
            />
          </svg>
        </button>
        <div
          class="sc-ftTHYK fReoLJ"
        >
          <p
            class="sc-iBYQkv diua-dV"
            style="color: rgb(1, 48, 89); font-weight: 700;"
          >
            Witaj User
          </p>
          <img
            alt="profile"
            class="sc-pyfCe jiCDTo"
            src="/src/assets/profile_picture.png"
          />
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-bqWxrE gOWFEp"
    >
      <button
        class="sc-eDvSVe bcqqiU"
        data-testid="side-menu-hamburger-icon"
      >
        <svg
          data-testid="hamburger-icon"
          fill="none"
          height="15"
          viewBox="0 0 45 23"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            stroke="#013059"
            stroke-width="3"
            x2="45"
            y1="21.5"
            y2="21.5"
          />
          <line
            stroke="#013059"
            stroke-width="3"
            x2="45"
            y1="11.5"
            y2="11.5"
          />
          <line
            stroke="#013059"
            stroke-width="3"
            x2="45"
            y1="1.5"
            y2="1.5"
          />
        </svg>
      </button>
      <div
        class="sc-ftTHYK fReoLJ"
      >
        <p
          class="sc-iBYQkv diua-dV"
          style="color: rgb(1, 48, 89); font-weight: 700;"
        >
          Witaj User
        </p>
        <img
          alt="profile"
          class="sc-pyfCe jiCDTo"
          src="/src/assets/profile_picture.png"
        />
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
